fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneDocumentFromRecord.java

```diff
@@ -183,7 +183,7 @@ public List<DocumentField> getFields() {
         public void addField(@Nonnull T source, @Nonnull String fieldName, @Nullable final Object value,
                              LuceneIndexExpressions.DocumentFieldType type,
                              boolean stored, boolean sorted,
-                             @Nonnull List<Integer> overriddenKeyRanges, int groupingKeyIndex,
+                             @Nonnull List<Integer> overriddenKeyRanges, int groupingKeyIndex, int keyIndex,
```

Comment:
Is the new `keyIndex` parameter actually used? I don't see a method that actually uses it any where

Reply:
it's used here: https://github.com/FoundationDB/fdb-record-layer/pull/1724/files#diff-11ef9c019db17a199d457b1610f89a129c31d03a5b14c006f27fd1f1454ba5e8R77

Reply:
I see... The idea being that it uses the machinery to translate the primary key data as well.

A lot of this code seems to be duplicating code in the partial record creation code in `fdb-record-layer-core`, but I suppose refactoring this is something that can be done at a later date